language-support/ts/daml-types/index.ts

```diff
@@ -5,13 +5,21 @@ import * as jtv from '@mojotech/json-type-validation';
 /**
  * Interface for companion objects of serializable types. Its main purpose is
  * to describe the JSON encoding of values of the serializable type.
+ *
+ * @typeparam T The contract template type.
  */
 export interface Serializable<T> {
-  // NOTE(MH): This must be a function to allow for mutually recursive decoders.
+  /**
+   * The decoder for a contract of template T.
+   *
+   * NB: This is a function to allow for mutually recursive decoders.
```

Comment:
That seems like an implementation detail. Do we want to keep this in the docs? In fact should we hide the whole `Serializable` thing?

Reply:
My thinking was that if someone is looking at the documentation of daml-types, then they probably want to build something on their own. So the serializable thing might be of interest to them. But I wouldn't mind either to make it internal.